src/Features/Core/Portable/SpellCheck/AbstractSpellCheckSpanService.cs

```diff
@@ -15,35 +16,39 @@
 
 namespace Microsoft.CodeAnalysis.SpellCheck
 {
-    internal abstract class AbstractSpellCheckSpanService : ISpellCheckSpanService
+    internal abstract class AbstractSpellCheckSpanService(char? escapeCharacter) : ISpellCheckSpanService
     {
+        private readonly char? _escapeCharacter = escapeCharacter;
+
         public async Task<ImmutableArray<SpellCheckSpan>> GetSpansAsync(Document document, CancellationToken cancellationToken)
         {
             var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            return GetSpans();
 
-            return GetSpans(document, root, cancellationToken);
-        }
-
-        private static ImmutableArray<SpellCheckSpan> GetSpans(Document document, SyntaxNode root, CancellationToken cancellationToken)
-        {
-            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
-            var classifier = document.GetRequiredLanguageService<ISyntaxClassificationService>();
-            var virtualCharService = document.GetRequiredLanguageService<IVirtualCharLanguageService>();
+            // Broken into its own method as it uses a ref-struct, which isn't allowed with the async call above.
```

Comment:
💭 `ref struct` seems unnecessary here since it's a `readonly struct`

Reply:
i don't want this going on the heap :)